server/app/services/program/ProgramService.java

```diff
@@ -223,27 +223,26 @@ public ImmutableList<ProgramDefinition> getAllProgramDefinitionVersions(long pro
   }
 
   private CompletionStage<ProgramDefinition> syncProgramAssociations(ProgramModel program) {
-    if (isActiveOrDraftProgram(program)) {
+    VersionModel activeVersion = versionRepository.getActiveVersion();
```

Comment:
**suggestion (non-blocking):** Consider putting this in a helper method. If the helper method is called something like `isDraftVersion()`, then no comment is required here.